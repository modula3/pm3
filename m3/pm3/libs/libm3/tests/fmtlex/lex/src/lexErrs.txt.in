# Last modified on Mon Dec 13 16:38:19 PST 1993 by heydon

# The examples in this file should all produce lexical errors. Most of the
# examples were created by examing the "Lex.m3" source file and picking inputs
# to test each error case.

# scan (matches non-blanks)
# There is no way for Lex.Scan to fail; it doesn't raise Lex.Error.

# match
m abc abde	# => "de"
m abcde abc	# => ""

# booleans
b tru		# => ""
b truXYZ	# => "XYZ"

# signed base-10 integers
i		# => ""
i + 5		# => " 5"
i -ABC		# => "ABC"

# signed integers (not base 10)
i -3_1234ABC	# => "ABC"
I 3 -1234ABC	# => "ABC"
i 9_789012ABC	# => "ABC"
I 9 789012ABC	# => "ABC"
i 11_789ABCXYZ	# => "XYZ"
I 11 789ABCXYZ	# => "XYZ"
i 5_ABC		# => "ABC"

# unsigned integers
u		# => ""
u -2		# => "-2"
u 5_ABC		# => "ABC"

# single-precision real numbers
r ..		# => "."
r .		# => ""

# double-precision real numbers

# extended-precision real numbers
