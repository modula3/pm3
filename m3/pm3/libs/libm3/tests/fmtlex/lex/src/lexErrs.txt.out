m abc abde	# => "de"
** Raised "Lex.Error"
-> "de"

m abcde abc	# => ""
** Raised "Lex.Error"
-> ""

b tru		# => ""
** Raised "Lex.Error"
-> ""

b truXYZ	# => "XYZ"
** Raised "Lex.Error"
-> "XYZ"

i		# => ""
** Raised "Lex.Error"
-> ""

i + 5		# => " 5"
** Raised "Lex.Error"
-> " 5"

i -ABC		# => "ABC"
** Raised "Lex.Error"
-> "ABC"

i -3_1234ABC	# => "ABC"
** Raised "Lex.Error"
-> "ABC"

I 3 -1234ABC	# => "ABC"
** Raised "Lex.Error"
-> "ABC"

i 9_789012ABC	# => "ABC"
** Raised "Lex.Error"
-> "ABC"

I 9 789012ABC	# => "ABC"
** Raised "Lex.Error"
-> "ABC"

i 11_789ABCXYZ	# => "XYZ"
** Raised "Lex.Error"
-> "XYZ"

I 11 789ABCXYZ	# => "XYZ"
** Raised "Lex.Error"
-> "XYZ"

i 5_ABC		# => "ABC"
** Raised "Lex.Error"
-> "ABC"

u		# => ""
** Raised "Lex.Error"
-> ""

u -2		# => "-2"
** Raised "Lex.Error"
-> "-2"

u 5_ABC		# => "ABC"
** Raised "Lex.Error"
-> "ABC"

r ..		# => "."
** Raised "Lex.Error"
-> "."

r .		# => ""
** Raised "Lex.Error"
-> ""

